Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt

```diff
@@ -20,15 +20,15 @@ fun Collection<CwaCovidCertificate>.toCertificateSortOrder(): List<CwaCovidCerti
             {
                 when (it) {
                     is VaccinationCertificate -> it.vaccinatedOn
-                    is TestCertificate -> it.sampleCollectedAt.toLocalDateUserTz()
+                    is TestCertificate -> it.sampleCollectedAt?.toLocalDateUserTz()
                     is RecoveryCertificate -> it.validFrom
                     else -> throw IllegalStateException("Can't sort $it")
                 }
             },
             {
                 when (it) {
                     is VaccinationCertificate -> it.headerIssuedAt.toLocalDateUserTz()
-                    is TestCertificate -> it.sampleCollectedAt.toLocalDateUserTz()
+                    is TestCertificate -> it.sampleCollectedAt?.toLocalDateUserTz()
```

Comment:
do we need to provide `default`  dates ?

Reply:
I added a unit test covering this scenario, a test without a date is moved to the end of the order